editor/src/ast/ast_forest.rs

```diff
@@ -25,6 +25,34 @@ impl<'l> AstForest<'l> {
         }
     }
 
+    pub fn new_tree_by_name(
+        &self,
+        language: &'l Language,
+        construct_name: &ConstructName,
+        notation_set: &'l NotationSet,
+    ) -> Option<Ast<'l>> {
+        let construct = language.lookup_construct(construct_name);
+        let node = Node {
+            bounds: Bounds::empty(),
+            language,
+            construct,
+            notation: notation_set.lookup(construct_name),
+        };
+        let ast = match &construct.arity {
+            Arity::Fixed(sorts) => {
+                let children = vec![self.new_hole_tree(language, notation_set).tree; sorts.len()];
+                Ast::new(self.forest.new_branch(node, children))
+            }
+            Arity::Flexible(_) => Ast::new(self.forest.new_branch(node, vec![])),
+            Arity::Text => {
+                let leaf = self.forest.new_leaf(Text::new_inactive());
+                Ast::new(self.forest.new_branch(node, vec![leaf]))
+            }
+            _ => unimplemented!(),
```

Comment:
Should have a comment saying what case is unimplemented. (Or better yet, match on it.)